src/main/java/User.java

```diff
@@ -0,0 +1,16 @@
+package src.main.java;
+
+public class User {
+	private ATMaccount account = new ATMaccount(this);
+	private String name;
+	public User(String name) {
+		this.name = name;
+	}
+	public ATMaccount getAccount() {
+		return account;
+	}
+	
+	public String getName() {
+		return name;
+	}
+}
```

Comment:
This `User` class has a few problems:

First, it violates the [Single Responsibility Principle](https://www.baeldung.com/java-single-responsibility-principle). The class is responsible for both managing user information (name) and handling account-related functionality. Ideally, each class should have a single responsibility. In this way you improve code organization and maintainability.

Second, by creating an `ATMAccount` instance directly inside the `User` class, you tightly couple the two classes. Changes to the `ATMaccount` class may affect the `User` class, and vice versa. It’s generally better to keep classes loosely coupled to improve maintainability and flexibility.

Third, the `account` field is eagerly initialized with a new `ATMaccount` instance in the constructor. This means that every `User` object will always have an associated `ATMaccount`, even if it’s not needed. [Eager initialization](https://www.infoworld.com/article/2336990/lazy-vs-eager-instantiation-in-java-which-is-better.html) can lead to unnecessary memory usage and performance overhead.

Reply:
It doesn't really handles the account related functionalities but just keep track of the account of the user. Or does it? If yes, how? And if that's not "conventional" how sould I try to keep track of it? 

Yeah I made that on purpose, I learned that at university and though it would be a good idea to link both, knowing each account has to be related to an user. So that we can make sure that this bond stays during the whole application.  We learned this [here ](https://btj.github.io/ogp-notes/entity_relationship_abstractions.html). If you think it's not a good thing to do would you like to explain me why? (I'm more the type of guy that I likes to know the why before I change something, so I don't just do what I'm told, but understand it first). 

I understand the third comment. I'll change that.